flink-python/src/main/java/org/apache/flink/api/common/python/PythonBridgeUtils.java

```diff
@@ -266,6 +278,92 @@ private static Object getPickledBytesFromJavaObject(Object obj, LogicalType data
 		}
 	}
 
+	public static Object getPickledBytesFromJavaObject(Object obj, TypeInformation<?> dataType) throws IOException {
+		Pickler pickler = new Pickler();
+		initialize();
+		if (obj == null) {
+			return new byte[0];
+		} else {
+			if (dataType instanceof SqlTimeTypeInfo) {
+				SqlTimeTypeInfo<?> sqlTimeTypeInfo = SqlTimeTypeInfo.getInfoFor(dataType.getTypeClass());
+				if (sqlTimeTypeInfo == DATE) {
+					long time;
+					if (obj instanceof LocalDate) {
+						time = ((LocalDate) (obj)).toEpochDay();
+					} else {
+						time = ((Date) obj).toLocalDate().toEpochDay();
+					}
+					return pickler.dumps(time);
+				} else if (sqlTimeTypeInfo == TIME) {
+					long time;
+					if (obj instanceof LocalTime) {
+						time = ((LocalTime) obj).toNanoOfDay();
+					} else {
+						time = ((Time) obj).toLocalTime().toNanoOfDay();
+					}
+					time = time / 1000;
+					return pickler.dumps(time);
+				} else if (sqlTimeTypeInfo == TIMESTAMP) {
+					if (obj instanceof LocalDateTime) {
+						return pickler.dumps(Timestamp.valueOf((LocalDateTime) obj));
+					} else {
+						return pickler.dumps(obj);
+					}
+				}
+			} else if (dataType instanceof RowTypeInfo) {
+				Row tmpRow = (Row) obj;
+				TypeInformation<?>[] tmpRowFieldTypes = ((RowTypeInfo) dataType).getFieldTypes();
+				List<Object> rowFieldBytes = new ArrayList<>(tmpRow.getArity() + 1);
+				rowFieldBytes.add(new byte[]{tmpRow.getKind().toByteValue()});
+				for (int i = 0; i < tmpRow.getArity(); i++) {
+					rowFieldBytes.add(getPickledBytesFromJavaObject(
+						tmpRow.getField(i),
+						tmpRowFieldTypes[i]));
+				}
+				return rowFieldBytes;
+			} else if (dataType instanceof MapTypeInfo) {
```

Comment:
Map type is still not supported in the Python DataStream API and so we should remove this.